app/services/doc_auth/error_generator.rb

```diff
@@ -252,7 +151,182 @@ def get_selfie_error(liveness_enabled, response_info)
+
+  class ErrorGenerator
+    attr_reader :config
+
+    # These constants are the key names for the TrueID errors hash that is returned
+    ID = :id
+    FRONT = :front
+    BACK = :back
+    SELFIE = :selfie
+    GENERAL = :general
+
+    ACCEPTED_ISSUER_TYPES = [DocAuth::LexisNexis::IssuerTypes::STATE_OR_PROVINCE.name,
+                             DocAuth::LexisNexis::IssuerTypes::UNKNOWN.name]
+
+    ERROR_KEYS = [
+      ID,
+      FRONT,
+      BACK,
+      SELFIE,
+      GENERAL,
+    ].to_set.freeze
+
+    ALERT_MESSAGES = {
+      '1D Control Number Valid': { type: BACK, msg_key: Errors::REF_CONTROL_NUMBER_CHECK },
+      '2D Barcode Content': { type: BACK, msg_key: Errors::BARCODE_CONTENT_CHECK },
+      '2D Barcode Read': { type: BACK, msg_key: Errors::BARCODE_READ_CHECK },
+      'Birth Date Crosscheck': { type: ID, msg_key: Errors::BIRTH_DATE_CHECKS },
+      'Birth Date Valid': { type: ID, msg_key: Errors::BIRTH_DATE_CHECKS },
+      'Control Number Crosscheck': { type: BACK, msg_key: Errors::CONTROL_NUMBER_CHECK },
+      'Document Classification': { type: ID, msg_key: Errors::ID_NOT_RECOGNIZED },
+      'Document Crosscheck Aggregation': { type: ID, msg_key: Errors::DOC_CROSSCHECK },
+      'Document Expired': { type: ID, msg_key: Errors::DOCUMENT_EXPIRED_CHECK },
+      'Document Number Crosscheck': { type: ID, msg_key: Errors::DOC_NUMBER_CHECKS },
+      'Expiration Date Crosscheck': { type: ID, msg_key: Errors::EXPIRATION_CHECKS },
+      'Expiration Date Valid': { type: ID, msg_key: Errors::EXPIRATION_CHECKS },
+      'Full Name Crosscheck': { type: ID, msg_key: Errors::FULL_NAME_CHECK },
+      'Issue Date Crosscheck': { type: ID, msg_key: Errors::ISSUE_DATE_CHECKS },
+      'Issue Date Valid': { type: ID, msg_key: Errors::ISSUE_DATE_CHECKS },
+      'Layout Valid': { type: ID, msg_key: Errors::ID_NOT_VERIFIED },
+      'Near-Infrared Response': { type: ID, msg_key: Errors::ID_NOT_VERIFIED },
+      'Photo Printing': { type: FRONT, msg_key: Errors::VISIBLE_PHOTO_CHECK },
+      'Physical Document Presence': { type: ID, msg_key: Errors::ID_NOT_VERIFIED },
+      'Sex Crosscheck': { type: ID, msg_key: Errors::SEX_CHECK },
+      'Visible Color Response': { type: ID, msg_key: Errors::VISIBLE_COLOR_CHECK },
+      'Visible Pattern': { type: ID, msg_key: Errors::ID_NOT_VERIFIED },
+      'Visible Photo Characteristics': { type: FRONT, msg_key: Errors::VISIBLE_PHOTO_CHECK },
+    }.freeze
+
+    SUPPORTED_ID_CLASSNAME = ['Identification Card', 'Drivers License'].freeze
+
+    def initialize(config)
+      @config = config
+    end
+
+    def generate_doc_auth_errors(response_info)
+      # when entered here, it's decided the doc auth is not successful
+
+      # scan unknown(handled) error, make sure `warn_notify` it
+      # if unhandled error found
+      unknown_fail_count = scan_for_unknown_alerts(response_info)
+
+      # check whether ID type supported
+      id_type_error = IdTypeErrorHandler.new.handle(response_info)
```

Comment:
```suggestion
      id_type_error_handler = IdTypeErrorHandler.new
      id_type_error = id_type_error_handler.handle(response_info)
```
i think this is more debuggable ... kind of like how you implemented # check selfie error